src/Module/Admin/Users.php

```diff
@@ -0,0 +1,318 @@
+			notice(L10n::tt('%s user unblocked', '%s users unblocked', count($users)));
+		}
+
+		if (!empty($_POST['page_users_delete'])) {
+			foreach ($users as $uid) {
+				if (local_user() != $uid) {
+					User::remove($uid);
+				} else {
+					notice(L10n::t('You can\'t remove yourself'));
+				}
+			}
+
+			notice(L10n::tt('%s user deleted', '%s users deleted', count($users)));
+		}
+
+		if (!empty($_POST['page_users_approve'])) {
+			require_once 'mod/regmod.php';
+			foreach ($pending as $hash) {
+				user_allow($hash);
+			}
+		}
+
+		if (!empty($_POST['page_users_deny'])) {
+			require_once 'mod/regmod.php';
+			foreach ($pending as $hash) {
+				user_deny($hash);
+			}
+		}
+
+		$a->internalRedirect('admin/users');
+	}
+
+	public static function content()
+	{
+		parent::content();
+
+		$a = self::getApp();
+
+		if ($a->argc > 3) {
+			// @TODO: Replace with parameter from router
+			$action = $a->argv[2];
+			$uid = $a->argv[3];
+			$user = DBA::selectFirst('user', ['username', 'blocked'], ['uid' => $uid]);
+			if (!DBA::isResult($user)) {
+				notice('User not found' . EOL);
+				$a->internalRedirect('admin/users');
+				return ''; // NOTREACHED
+			}
+
+			switch ($action) {
+				case 'delete':
+					if (local_user() != $uid) {
+						parent::checkFormSecurityTokenRedirectOnError('/admin/users', 'admin_users', 't');
+						// delete user
+						User::remove($uid);
+
+						notice(L10n::t('User "%s" deleted', $user['username']));
+					} else {
+						notice(L10n::t('You can\'t remove yourself'));
+					}
+					break;
+				case 'block':
+					parent::checkFormSecurityTokenRedirectOnError('/admin/users', 'admin_users', 't');
+					DBA::update('user', ['blocked' => 1], ['uid' => $uid]);
+					notice(L10n::t('User "%s" blocked', $user['username']));
+					break;
+				case 'unblock':
+					parent::checkFormSecurityTokenRedirectOnError('/admin/users', 'admin_users', 't');
+					DBA::update('user', ['blocked' => 0], ['uid' => $uid]);
+					notice(L10n::t('User "%s" unblocked', $user['username']));
+					break;
+			}
+
+			$a->internalRedirect('admin/users');
+		}
+
+		/* get pending */
+		$pending = Register::getPending();
+
+		$pager = new Pager($a->query_string, 100);
+
+		/* ordering */
+		$valid_orders = [
+			'contact.name',
+			'user.email',
+			'user.register_date',
+			'user.login_date',
+			'lastitem_date',
+			'user.page-flags'
+		];
+
+		$order = 'contact.name';
+		$order_direction = '+';
+		if (!empty($_GET['o'])) {
+			$new_order = $_GET['o'];
+			if ($new_order[0] === '-') {
+				$order_direction = '-';
+				$new_order = substr($new_order, 1);
+			}
+
+			if (in_array($new_order, $valid_orders)) {
+				$order = $new_order;
+			}
+		}
+		$sql_order = '`' . str_replace('.', '`.`', $order) . '`';
+		$sql_order_direction = ($order_direction === '+') ? 'ASC' : 'DESC';
+
+		$usersStmt = DBA::p("SELECT `user`.*, `contact`.`name`, `contact`.`url`, `contact`.`micro`, `user`.`account_expired`, `contact`.`last-item` AS `lastitem_date`

```

Comment:
we can define a `Model\User::getUsers($start, $count, $order = 'contact.name', $order_direction = '+')`

```php
public static function getUsers($start, $count, $order = 'contact.name', $order_direction = '+')
{
	/* ordering */
	$valid_orders = [
		'contact.name',
		'user.email',
		'user.register_date',
		'user.login_date',
		'lastitem_date',
		'user.page-flags'
	];
	
	$sql_order = '`' . str_replace('.', '`.`', $order) . '`';
	$sql_order_direction = ($order_direction === '+') ? 'ASC' : 'DESC';
	$usersStmt = DBA::p("SELECT `user`.*, `contact`.`name`, `contact`.`url`, `contact`.`micro`, `user`.`account_expired`, `contact`.`last-item` AS `lastitem_date`
FROM `user`
			INNER JOIN `contact` ON `contact`.`uid` = `user`.`uid` AND `contact`.`self`
			WHERE `user`.`verified`
			ORDER BY $sql_order $sql_order_direction LIMIT ?, ?", $start, $count
	);
	return DBA::toArray($usersStmt);
}
```

Reply:
Yes, another time.